xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-test/xwiki-platform-ckeditor-test-pageobjects/src/main/java/org/xwiki/ckeditor/test/po/QuickActionDropdown.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.ckeditor.test.po;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebElement;
+import org.xwiki.stability.Unstable;
+import org.xwiki.test.ui.po.BaseElement;
+
+
+/**
+ * Models a Quick Action drop-down.
+ *
+ * @version $Id$
+ * @since 15.5RC1
+ */
+@Unstable
+public class QuickActionDropdown extends BaseElement
+{
+    
+    /**
+     * The associated RichTextArea instance.
+     */
+    private final RichTextAreaElement textArea;
+    
+    /**
+     * The associated selected element id.
+     */
+    private String selectedId;
+    
+    /**
+     * Creates a new Quick Action drop-down instance.
+     * 
+     * @param editor - CKEditor instance
+     */
+    public QuickActionDropdown(CKEditor editor)
+    {
+        this.textArea = editor.getRichTextArea();
+    }
+    
+    /**
+     * Finds the dropdown.
+     * @return the dropdown element
+     */
+    public WebElement getDropdown()
+    {
+        try {
+            return getDriver().findElement(By.cssSelector(
+                ".cke_autocomplete_opened .cke_autocomplete_selected"));
+        } catch (NoSuchElementException e) {
+            return null;
+        }
+    }
+    
+    
+    /**
+     * Open the Quick Action drop-down.
+     * 
+     */
+    public void open()
```

Comment:
I'm wondering if it's not better to have a ``CKEditor.openQuickActions()`` method instead. This would allow you to do the wait in the ``QuickActionDropdown`` constructor and to also get a reference to the underlying ``WebElement`` instance there.

Reply:
I moved it as you suggested, but I'm not sure it is necessary since as you mentioned, the user does not wait for the dropdown to show